sparrow-py/pysrc/sparrow_py/expr.py

```diff
@@ -0,0 +1,209 @@
+"""Defines classes representing Kaskada expressions."""
+
+import sys
+from typing import Callable
+from typing import Tuple
+from typing import Union
+from typing import final
+
+from sparrow_py import ffi
+
+
+class Expr(object):
+    """A Kaskada expression."""
+
+    ffi: ffi.Expr
+
+    def __init__(self, name: str, *args: "Expr") -> None:
```

Comment:
One way to look at this is the less versions we support the less work. The other way is that we might have clients that are not going to be able to easily upgrade to use our tools. So ... decisions, decisions. :)

Reply:
Agreed. I think for now, >=3.8 is nice. It's relatively minimal:

If we moved to >=3.10 we could use slightly simpler type annotations. (`Expr` instead of `"Expr"`). This doesn't seem worth it.

If we moved to >=3.11 we could be confident we produce better errors. This seems pretty weak -- the user can choose to use the version that produces better errors if they want better errors.